librz/core/cmd/cmd_analysis.c

```diff
@@ -5639,6 +5639,31 @@ static inline RzCmdStatus graph_handler(RzCore *core, RzCoreGraphType type, RzCo
 	return RZ_CMD_STATUS_OK;
 }
 
+static inline RzCoreGraphFormat graph_format_from_string(const char *x) {
+	if (strncmp(x, "ascii", sizeof("ascii")) == 0 || RZ_STR_ISEMPTY(x) || x[0] == ' ') {
+		return RZ_CORE_GRAPH_FORMAT_ASCII_ART;
+	} else if (strncmp(x, "cmd", sizeof("cmd")) == 0 || x[0] == '*') {
+		return RZ_CORE_GRAPH_FORMAT_CMD;
+	} else if (strncmp(x, "dot", sizeof("dot")) == 0 || x[0] == 'd') {
+		return RZ_CORE_GRAPH_FORMAT_DOT;
+	} else if (strncmp(x, "gml", sizeof("gml")) == 0 || x[0] == 'g') {
+		return RZ_CORE_GRAPH_FORMAT_GML;
+	} else if (strncmp(x, "json", sizeof("json")) == 0 || x[0] == 'j') {
+		return RZ_CORE_GRAPH_FORMAT_JSON;
+	} else if (strncmp(x, "json-disasm", sizeof("json-disasm")) == 0 || x[0] == 'J') {
+		return RZ_CORE_GRAPH_FORMAT_JSON_DISASM;
+	} else if (strncmp(x, "sdb", sizeof("sdb")) == 0 || x[0] == 'k') {
+		return RZ_CORE_GRAPH_FORMAT_SDB;
+	} else if (strncmp(x, "tiny", sizeof("tiny")) == 0 || x[0] == 't') {
+		return RZ_CORE_GRAPH_FORMAT_TINY;
+	} else if (strncmp(x, "interactive", sizeof("interactive")) == 0 || x[0] == 'v') {
+		return RZ_CORE_GRAPH_FORMAT_VISUAL;
+	}
+
+	rz_warn_if_reached();
```

Comment:
not sure if this should be here or not.

Reply:
I think it's needed, it looks like rzshell doesn't detect the value of choices

![image](https://user-images.githubusercontent.com/24983369/185447982-84dfeb5e-c81c-4c0a-b04b-2bb10a66a7b8.png)

Reply:
@ret2libc what do you think

Reply:
i think it should be handled differently, without `rz_warn_if_reached`, but yes, choices are not enforced at rzshell level, they are just suggested (for now at least..)